crates/miden-tx/src/tests/kernel_tests/test_fpi.rs

```diff
@@ -654,50 +659,66 @@ fn test_fpi_execute_foreign_procedure() {
 // NESTED FPI TESTS
 // ================================================================================================
 
-/// Test the correctness of the nested foreign procedures call.
+/// Test the correctness of the cyclic foreign procedure calls.
 ///
 /// It checks that the account data pointers are correctly added and removed from the account data
 /// stack.
 ///
 /// The call chain in this test looks like so:
-/// `Native -> First FA -> Second FA`
+/// `Native -> First FA -> Second FA -> First FA`
 #[test]
-fn test_simple_nested_fpi() {
+fn test_nested_fpi_cyclic_invocation() {
     // ------ SECOND FOREIGN ACCOUNT ---------------------------------------------------------------
-    let storage_slots = vec![AccountStorage::mock_item_0().slot];
     let second_foreign_account_code_source = "
-        use.miden::account
+        use.miden::tx
+        
+        use.std::sys
 
-        export.get_item_foreign
-            # make this foreign procedure unique to make sure that we invoke the procedure of the 
-            # foreign account, not the native one
-            push.1 drop
-            exec.account::get_item
+        export.second_account_foreign_proc
+            # get the storage item at index 0
+            # pad the stack for the `execute_foreign_procedure` execution
```

Comment:
We're not actually accessing this account's storage here. Should we add that? We currently add a constant `10` to the return value of `get_item_foreign`. We could instead store `10` in the account slot and access that.

Reply:
Good point! I forgot to add the slot index, but because we were accessing the 0'th one it worked. I'll also change in to be the first slot to make sure that we are accessing the correct slot.